services/wallet/saved_addresses.go

```diff
@@ -82,8 +83,18 @@ func (sam *SavedAddressesManager) GetRawSavedAddresses() ([]SavedAddress, error)
 }
 
 func (sam *SavedAddressesManager) upsertSavedAddress(sa SavedAddress, tx *sql.Tx) error {
-	sqlStatement := "INSERT OR REPLACE INTO saved_addresses (address, name, favourite, removed, update_clock, chain_short_names, ens_name, is_test) VALUES (?, ?, ?, ?, ?, ?, ?, ?)"
-	var err error
+	savedAddresses, err := sam.GetSavedAddresses()
```

Comment:
@alaibe if the transaction `tx` has already started (not nil), you will get stuck here. Would be great if you could get saved addresses within transaction (otherwise new one may be added/removed) and also would be good to remove `else` block from condition in the line `99` and do insert or replace under transaction only, cause that insertion now depends on `CreatedAt`.